packages/headless/src/controllers/smart-snippet/headless-smart-snippet-interactive-inline-links.ts

```diff
@@ -0,0 +1,142 @@
+import {questionAnswering, search} from '../../app/reducers';
+import {SearchEngine} from '../../app/search-engine/search-engine';
+import {QuestionAnsweringSection} from '../../state/state-sections';
+import {loadReducerError} from '../../utils/errors';
+import {
+  buildInteractiveResultCore,
+  InteractiveResultCore,
+} from '../core/interactive-result/headless-core-interactive-result';
+import {
+  logOpenSmartSnippetInlineLink,
+  logOpenSmartSnippetSuggestionInlineLink,
+} from '../../features/question-answering/question-answering-analytics-actions';
+import {getObjectHash} from '../../utils/utils';
+
+export interface InlineLink {
+  linkText: string;
+  linkURL: string;
+}
+
+/**
+ * @internal
+ */
+export interface SmartSnippetInteractiveInlineLinksOptions {
+  selectionDelay?: number;
+}
+
+/**
+ * @internal
+ */
+export interface SmartSnippetInteractiveInlineLinksProps {
+  options?: SmartSnippetInteractiveInlineLinksOptions;
+}
+
+/**
+ * @internal
+ */
+export interface SmartSnippetInteractiveInlineLinks {
```

Comment:
Not sure why the Link is plural here, confused me a bit off the bat. It's multiple manners to interact with a link

Reply:
This controller handles all the links for a given smart snippet, that's why it's plural.